consensus/safety-rules/src/safety_rules.rs

```diff
@@ -27,26 +33,38 @@ use libra_types::{
 /// set)
 pub struct SafetyRules {
     persistent_storage: PersistentSafetyStorage,
-    validator_signer: ValidatorSigner,
+    validator_signer: Option<ValidatorSigner>,
     validator_verifier: Option<ValidatorVerifier>,
+    author: Author,
 }
 
 impl SafetyRules {
     /// Constructs a new instance of SafetyRules with the given persistent storage and the
     /// consensus private keys
     /// @TODO replace this with an API that takes in a SafetyRulesConfig
+    /// @TODO change the constructor to fit Option<validator_signer> and address the circular
+    /// dependency with reconcile_key() during initializationg
     pub fn new(author: Author, persistent_storage: PersistentSafetyStorage) -> Self {
-        let consensus_key = persistent_storage
-            .consensus_key()
-            .expect("Unable to retrieve consensus private key");
-        let validator_signer = ValidatorSigner::new(author, consensus_key);
         Self {
             persistent_storage,
-            validator_signer,
+            validator_signer: None,
             validator_verifier: None,
+            author,
```

Comment:
if you want to do this here, we should follow up with another PR, but this is good for now.